bin/metrics.js

```diff
@@ -0,0 +1,185 @@
+  )
+}
+
+function run (config, now) {
+  now = now || new Date()
+  var lastMidnight = Date.UTC(
+    now.getUTCFullYear(),
+    now.getUTCMonth(),
+    now.getUTCDate(),
+    0, 0, 0, 0
+  )
+  var os = require('os')
+  var log = require('../lib/logging')('bin.metrics')
+  var mysql = require('../lib/db/mysql')(log, require('../fxa-auth-db-server').errors)
+  var P = require('../lib/promise')
+  var self = this
+
+  return mysql.connect({
+    master: {
+      host: config.General.db_dnsname,
+      user: config.General.db_username,
+      password: config.General.db_password,
+      database: config.General.db_name
+    },
+    slave: {
+      host: config.General.db_dnsname,
+      user: config.General.db_username,
+      password: config.General.db_password,
+      database: config.General.db_name
+    },
+    patchKey: 'schema-patch-level'
+  })
+  .then(function (db) {
+    return db.readMultiple([
+      { sql: 'SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED' },
+      metricsQuery('countAccounts'),
+      metricsQuery('countVerifiedAccounts'),
+      metricsQuery('countAccountsWithTwoOrMoreDevices'),
+      metricsQuery('countAccountsWithThreeOrMoreDevices'),
+      metricsQuery('countAccountsWithMobileDevice'),
+    ], { sql: 'SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED' })
+  })
+  .then(function (results) {
+    assertResults(results, 1, 5)
+    fs.appendFileSync('/media/ephemeral0/fxa-admin/basic_metrics.log', JSON.stringify({
+      hostname: os.hostname(),
+      pid: process.pid,
+      op: 'account_totals',
+      total_accounts: results[1][0].count,
+      total_verified_accounts: results[2][0].count,
+      total_accounts_with_two_or_more_devices: results[3][0].count,
+      total_accounts_with_three_or_more_devices: results[4][0].count,
+      total_accounts_with_mobile_device: results[5][0].count,
+      time: (new Date(lastMidnight)).toISOString(),
+      v: 0
+    }))
+  })
+  .catch(function (error) {
+    log.error({ op: 'metrics.run', err: error })
+  })
+
+  function metricsQuery (queryName) {
+    return { sql: self[queryName], params: [ lastMidnight ] }
+  }
+}
+
+function assertResults (results, firstIndex, lastIndex) {
+  results.filter(function (result, index) {
+    return index >= firstIndex && index <= lastIndex
+  }).forEach(assertResult)
+}
+
+function assertResult (result) {
+  if (Array.isArray(result) && result.length === 1 && result[0].count >= 0) {
+    return
+  }
+
+  throw new Error('metrics query error')
+}
+
+function trim (string) {
+  return string.trim()
+}
+
+// The next two functions comprise a very rudimentary
+// parser for the following config file format:
+// [General]
+// db_dnsname: foo
+// db_username: bar
+// db_password: baz
+// db_name: qux
+function filterConfig (line) {
+  return line && line[0] !== '#' && line[0] !== ';'
+}
+
+function reduceConfig (parsed, line) {
+  var matches = line.match(/^\[(.+)\]$/)
+
+  if (matches) {
+    section = matches[1]
```

Comment:
I'm not thrilled about using a global variable for this.  If you add a higher-level `parseConfigFile` function, you could scope it to that function and put these helpers inside it like so:

```
function parseConfigFile(..) {
  var currentSection;

  function reduceConfig(...) {
  }

  return trim/filter/reduce/whatever
}